### PR TITLE
Make `stop_all` behave as `stop_self` on all actors

### PR DESCRIPTION
**Pull request by Restioson**
~~This PR introduces the `Controller` type. It is  like `Context`, but designed for supervisors to keep in order to spawn more actors on the same address. The difference is that it doesn't have a broadcast mailbox, so it doesn't cause the broadcast channel to fill up to the bound, as a supervisor's context is never actually used to run an actor.~~ *Moved to separate PR*

I changed `stop_all` behaviour to bring it in line with `stop_self` across all actors. I previously incorrectly tried to do this, but forgot that `stop_all` immediately disconnects the channel. I assumed this would be fine, but whilst working on itchysats's xtras supervisor I realised that `stop_all` would prevent the actor from ever being restarted. I am not sure this immediate disconnecting of the channel is desirable behaviour anyway, since in non-supervised actors, the actor's receivers will drop and the channel will disconnect that way on its own.

~~I also discovered an edge case bug in `try_advance_broadcast_tail` and added a test for it. This would allow a _new_ broadcast message to be sent in, even if the queue was already full, if one message was taken off with another waiting sender already present. The fix was quite simple and could be backported to master.~~ *Backported to master*

## TBD
- [x] Evaluate naming - I am not 100% convinced by it
- [x] Update docs

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Would you mind splitting this into more PRs?

The bugfix and alignment are uncontroversial but the Controller feature needs some more discussion I think: We could for example also solve this by disabling broadcast for the creation of the context.

Another idea is to have the supervisor run a custom event loop and thus I think there isn't an unused context around then?

**Comment by Restioson** [comment]
I've backported the fix into master so it's no longer just in this PR. I can split the controller bit out into a different PR too; I will probably remove it from this PR and add it to another, repurposing this for stop_all alignment

**Comment by Restioson** [comment]
Pushed now

File: src/context.rs [hunk 1]
```diff
@@ -85,10 +85,13 @@ impl<A: Actor> Context<A> {
 
     /// Stop all actors on this address. This is similar to [`Context::stop_self`] but it will stop
     /// all actors on this address.
+    ///
+    /// Unlike in previous versions of xtra, it does not disconnect the address immediately - it is
```
Comment (thomaseizinger) [inline src/context.rs:89]: I think this belongs in the changelog :)

**Review by thomaseizinger — APPROVED** [review]
I think this makes sense.

I am trusting the tests, didn't verify the implementation in detail!

File: src/context.rs [hunk 2]
```diff
@@ -85,10 +85,13 @@ impl<A: Actor> Context<A> {
...
+    /// equivalent to running [`Context::stop_self`] on all actors.
```
[the remark this answers is not in the document]
Comment (Restioson) [reply src/context.rs]: I think #96 could be good, but can be another PR ?

[the remark this answers is not in the document]
Comment (thomaseizinger) [reply src/context.rs]: Sure!

**Comment by thomaseizinger** [comment]
I took the freedom to change the PR title to a more descriptive one in the spirit of better commit messages :)

**Comment by Restioson** [comment]
Thanks! Do you think this is ready for merge?
